token_source.go

```diff
@@ -0,0 +1,214 @@
+package conductoroneapi
+
+import (
+	"bytes"
+	"context"
+	"crypto/ed25519"
+	"encoding/base64"
+	"errors"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
```

Comment:
at an SDK level, i wonder if we should align to `slog` in go 1.21, rather than ctxzap? or we need a way to inject getting a logger given a context.  I don't love pulliing large dependency trees like go-grpc-middleware

Reply:
or maybe we should just remove the logging from this layer and be better at returning good errors?

Reply:
I'm up for good errors. I mean, we should probably do a pass of the layers in our rpc / controller methods anyways